MobileFuseAdapter/src/main/java/com/chartboost/mediation/mobilefuseadapter/MobileFuseAdapter.kt

```diff
@@ -115,8 +115,24 @@ class MobileFuseAdapter : PartnerAdapter {
     ): Result<Unit> {
         PartnerLogController.log(SETUP_STARTED)
 
-        MobileFuse.initSdkServices(context)
-        return Result.success(PartnerLogController.log(SETUP_SUCCEEDED))
+        return suspendCancellableCoroutine { continuation ->
+            fun resumeOnce(result: Result<Unit>) {
+                if (continuation.isActive) {
+                    continuation.resume(result)
+                }
+            }
+
+            MobileFuse.init(object : SdkInitListener {
+                override fun onInitSuccess() {
+                    resumeOnce(Result.success(PartnerLogController.log(SETUP_SUCCEEDED)))
+                }
+
+                override fun onInitError() {
+                    resumeOnce(Result.success(PartnerLogController.log(SETUP_FAILED)))
```

Comment:
Let's pass a `Result.failure` here.
```suggestion
PartnerLogController.log(SETUP_FAILED)
resumeOnce(
    Result.failure(
        ChartboostMediationAdException(ChartboostMediationError.CM_INITIALIZATION_FAILURE_UNKNOWN)
    )
)
```

Reply:
Thanks, I added this change